### PR TITLE
Sample Exports: Format Filtering UI

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
This PR adds the UI to filter sample export files by format.

## Screenshots or screen recordings
image

## How to set up and validate locally
1. Through the UI, navigate to a group or project samples table and select samples with attachments of multiple formats
2. Click the `Create Export` dropdown and select `Sample Export`
3. In the dialog, verify the following:
- The sample accordion header contains `Samples (X selected)`. Ensure the correct number of samples is displayed
- Open the accordion and ensure the export description/explanation and sample listing are present
- Under file formats, use the `Add all` and `Remove all` buttons. Ensure they are disabled after use. Ensure the `Submit` button is also disabled when the `Selected` list contains no values.
4. Submit the export with a format(s), ensuring you're excluding/not selecting certain formats contained by the selected samples' attachments
5. After 30s on the export's `show` page, download the export and ensure the export files, the `manifest.json` contents, and the `Preview` tab only contain the expected files (ie: formats that were not selected should not be present).
6. Repeat the above with all formats and formats the samples do not have. Ensure all files and no files are included in the export, respectively.

Other tests:
7. Start a `Linelist Export`, and ensure the sample accordion header contains `Samples (X selected)` and that the correct number of samples is displayed. The export description was also moved into the accordion. Open the accordion and verify the description is present.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -1705,38 +1710,41 @@ en:
       tabs:
         summary: Summary
         preview: Preview
+    new:
+      samples_count:
+        zero: "Samples (Count: 0 samples)"
```
Comment (joshsadam) [inline config/locales/en.yml:1715]: **nitpick:** Can we reword this to `Samples (X selected)`

File: app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js [hunk 2]
```diff
@@ -20,33 +17,51 @@ export default class extends Controller {
   connect() {
     this.availableList = document.getElementById(this.availableListValue)
```
Comment (joshsadam) [inline app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js:18]: **suggestion:** Wouldn't these be better as stimulus targets?  That way you would not have to query the DOM to get them.

File: app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js [hunk 3]
```diff
@@ -20,33 +17,51 @@ export default class extends Controller {
...
     this.selectedList = document.getElementById(this.selectedListValue)
-    this.fullListItems = this.availableList.querySelectorAll("li")
+    this.allListItems = this.#constructAllListItems(this.availableList, this.selectedList)
+    this.#setInitialSelectAllState(this.availableList, this.addAllTarget)
+    this.#setInitialSelectAllState(this.selectedList, this.removeAllTarget)
     this.selectedList.addEventListener("mouseover", () => { this.#checkButtonStates() })
```
Comment (joshsadam) [inline app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js:23]: **nitpick:** If you create event listeners in a `connect`, removing them in the `disconnect` is good practice.

**Review by joshsadam — CHANGES_REQUESTED** [review]
This is looking really good, just a couple comments.  One thought, is is possible to restrict what is available to include based on what the sample actually has?  By this I mean if id doesn't have a genbank file, can we hide that option?

**Comment by ericenns** [comment]
> This is looking really good, just a couple comments. One thought, is is possible to restrict what is available to include based on what the sample actually has? By this I mean if id doesn't have a genbank file, can we hide that option?

This may be a nice idea for the future, but right now to determine this we would have to query all the samples select to reduce the list which currently can't be done quickly due to missing indexing etc. So for now, lets not reduce the list of formats for this iteration.

File: config/locales/en.yml (hunk 1 above)
Comment (ChrisHuynh333) [reply config/locales/en.yml]: Fixed in 3d10793efd87aa2ebb37a138a403215178554bbb

File: app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js [hunk 4]
```diff
@@ -20,33 +17,51 @@ export default class extends Controller {
...
-    const selected_metadata = this.selectedList.querySelectorAll("li")
-    const available_metadata = this.availableList.querySelectorAll("li")
-    if (selected_metadata.length == 0) {
+    const selected_values = this.selectedList.querySelectorAll("li")
+    const available_values = this.availableList.querySelectorAll("li")
+    if (selected_values.length == 0) {
       this.#setSubmitButtonDisableState(true)
       this.#setAddOrRemoveButtonDisableState(this.removeAllTarget, true)
       this.#setAddOrRemoveButtonDisableState(this.addAllTarget, false)
-    } else if (available_metadata.length == 0) {
```
[the remark this answers is not in the document]
Comment (ChrisHuynh333) [reply app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js:49]: Fixed in 3d10793efd87aa2ebb37a138a403215178554bbb

File: app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js (hunk 3 above)
Comment (ChrisHuynh333) [reply app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js:23]: Fixed in 3d10793efd87aa2ebb37a138a403215178554bbb

File: app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js (hunk 2 above)
Comment (ChrisHuynh333) [reply app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js:18]: As discussed, we'll leave this as is

File: app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js (hunk 3 above)
Comment (joshsadam) [inline app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js:23]: **nitpick:** Just noticed this.  Change this in the connect and disconnect methods:
```javascript
this.selectedList.addEventListener("mouseover", () => { this.#checkButtonStates() })
```
to
```javascript
this.selectedList.addEventListener("mouseover", this.#checkButtonStates);
```
The remove actually won't work properly the way you have it.

Comment (joshsadam) [inline app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js:23]: **nitpick:** Just noticed this.  Change this in the connect and disconnect methods:
```javascript
this.selectedList.addEventListener("mouseover", () => { this.#checkButtonStates() })
```
to
```javascript
this.selectedList.addEventListener("mouseover", this.#checkButtonStates);
```
The remove actually won't work properly the way you have it.

Comment (ChrisHuynh333) [reply app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js]: I lost functionality doing the above so after a little research, handled it like so: 4fff61f48597e8a014ef8b8976a206e42edc9d61

**Review by joshsadam — APPROVED** [review]
Looks good to me

**Review by joshsadam — APPROVED** [review]
